### PR TITLE
Integrate HQMTool into existing infrastructure

### PR DESCRIPTION
**Pull request by sloede**

### CODE REVIEW

**Comment by sloede** [comment]
For the docs:

iceCreamCone

**Comment by sloede** [comment]
@DavidAKopriva I think this is ready to merge into your branch, if you agree. **Afterwards**, you can use the HQMTool functionality as follows:

Clone this repo, enter it, and switch to your branch:
```shell
git clone git@github.com:trixi-framework/HOHQMesh.jl.git
cd HOHQMesh.jl
git checkout InteractiveTool
```
Then, start Julia with
```julia
julia --project=.
```

Load the HOHQMesh package by executing
```julia
julia> using Revise, HOHQMesh
```
(omit Revise if you do not want to use it, it is nice for development but optional for usage)

Finally, you can access all newly added functions by prefixing their names with `HOHQMesh.`, e.g.,
```julia
julia> HOHQMesh.runDemo()
```

I tried to leave everything else intact such that you can keep on developing. Specifically,
* `HQMTool.jl` and its associated folders now live under `src/`
* The documentation lives under `docs/src/` and can be accessed at https://trixi-framework.github.io/HOHQMesh.jl/previews/PR10 (again, *after* this PR has been merged into your branch)

A quick look through the code told me that there might be some opportunities for cleanup/speedups/improvements, but features come first. Thus I'd focus on getting everything ready in terms of functionality first.

If you have questions, please let me know.

**Review by DavidAKopriva — APPROVED** [review]
OK. Merge OK.

**Pull request merged**
